app/views/hyrax/file_sets/_actions.html.erb

```diff
@@ -39,18 +39,26 @@
                       data: { label: file_set.id } %>
         </li>
 
-        <% if file_set.pdf? %>
+        <% if Flipflop.enabled?(:annotation) && file_set.pdf? %>
           <li role="menuitem" tabindex="-1">
             <%= link_to t('.read_and_annotate'), hyku_addons.pdf_viewer_path(hyrax.download_url(file_set.id).split('/')&.last&.split('?')&.first),  { title: 'Annotate', id: "file_read", target: "_blank" } %>
           </li>
         <% end %>
       <% end %>
     </ul>
   </div>
+<% elsif Flipflop.enabled?(:annotation) && file_set.pdf? %>
+  <%= link_to t('.read_and_annotate'), 
+  hyku_addons.pdf_viewer_path(hyrax.download_url(file_set.id).split('/')&.last&.split('?')&.first),  
+  title: 'Annotate', 
+  id: "file_read", 
+  target: "_blank",
+  class: 'btn btn-default btn-sm' %>
 <% elsif can?(:download, file_set.id) %>
```

Comment:
can?(:download, file_set.id) in line 57 is a repeat of line 32, I am trying to understand why we need hem in both places?

Reply:
Secondly if it turns out that we need it in both places, should it move into  a partial for sharing?

Reply:
@mankind keep in mind that using partials is closer than keeping everything in the same place.:)

Reply:
One of the functions of this partial is to decide whether to render a dropdown list or a single button.  If there are multiple conditions you fulfil then we need a dropdown.  There are only two circumstances where you can only fulfil one: when you can only annotate or only download.  Once we've decided that a dropdown is needed we still need to check which options to fill it with, which is why there are individual checks matching each of the conditions.  I will refactor this class to make this more clear, as I have tried to get this right several times and it still remains obscure.

Reply:
Ok thanks @thenapking  for the clarification.

Reply:
Addressed below.